packages/primevue/src/inputnumber/InputNumber.vue

```diff
@@ -401,14 +402,20 @@ export default {
                     break;
 
                 case 'ArrowLeft':
-                    if (!this.isNumeralChar(inputValue.charAt(selectionStart - 1))) {
+                    if (selectionRange > 1) {
```

Comment:
Previously, this switch case block was behaving as follows: if the character to the left of the current cursor position is not a numeral (or a valid part of the number format, such as a decimal point, grouping character, or minus sign), is to prevent the default action (cursor movement), This ensures that the user cannot accidentally move the cursor to the prefix for example. 

However, this logic did not account for cases where there is a text selection, often occurring when users navigate between fields using the TAB key. In such cases, selectionStart would be 0, which coincides with the prefix (if provided). This resulted in the ArrowLeft and ArrowRight keys not working as expected, leaving users unable to properly adjust the cursor position.